### PR TITLE
fix: improve thor slippage

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Use the recommended slippage from the thor quote endpoint when calculating the memo.

This feeds into the limit value.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A - prod issue

## Risk

High - touches memos!

## Testing

Ensure that the amount we show the user in the "amount after slippage part of the UI", and the slippage used to calculate the memo is the one returned from the thor quote endpoint. It is also a function of if it was a streaming swap or not.

The limit is calculated by: expected amount - recommended slippage.

### Engineering

Check the actual memo that we put into the TX.

### Operations

☝️

## Screenshots (if applicable)

N/A

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5181** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5181" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5181?utm_source=stack-comment).

**Review by gomesalexandre — COMMENTED** [review]
First pass, functionally does what it says on the box ✅ 

### Regular THOR swaps (i.e non-streaming)

#### Smallish trade actually executed

- Limit matches actual expected amount out (expected amount - slippage), and slippage matches THOR-returned one

image
image
image

i.e 2.00127-(2.00127*1/100) = 1.9812573

Note how we use 1 as a safety slippage as THOR returned 0 slippage here

- Limit in memo matches our expected amount out after slippage

image

#### Biggish trade, not actually executed for testing funds reasons

- Limit matches actual expected amount out (expected amount - slippage), and slippage matches THOR-returned one

image
image

i.e 11562.27377451-11562.27377451*(33.23/100) = 7720.13019924

### Streaming swaps

#### Largish trade because streaming swaps - not executed

image

i.e 25,567.07342886-(25,567.07342886*0.05/100) = 25554.2898921

File: src/lib/swapper/swappers/ThorchainSwapper/getThorTradeQuote/getTradeQuote.ts [hunk 1]
```diff
@@ -384,11 +388,18 @@ export const getThorTradeQuote = async (
...
-            slippageBps,
+            slippageBps: estimatedSlippageBps,
             expectedAmountOutThorBaseUnit,
             isStreaming,
             estimatedExecutionTimeMs,
-          }) => {
+          }): ThorTradeQuote => {
+            /*
+             We start with the expected slippage amount, as given to us by the THORSwap API for the current quote.
+              We then add 1% to allow for market movement, which prevents failed/refunded trades.
```
Comment (gomesalexandre) [inline src/lib/swapper/swappers/ThorchainSwapper/getThorTradeQuote/getTradeQuote.ts:398]: ✅ as confirmed by testing!

**Review by gomesalexandre — APPROVED** [review]
Looks good to me both from an implementation and a functional perspective, this does what it says on the box. Stellar PR!

**Comment by 0xApotheosis** [comment]
Still thinking through this, not yet ready to merge.

**Comment by 0xApotheosis** [comment]
Closing this, as recommended slippage is actually incorrect terminology. Whilst this PR does what it intended to do, we don't actually want to consider the amount from the quote API in the limit because, after checking with THORChain devs, this amount is _already_ included in the `expected_amount_out` that we get back from the response. 

@gomesalexandre FYI, and thank you for your stellar review 🙏

**Pull request closed without merging**